network/p2p/router_test.go

```diff
@@ -27,6 +27,11 @@ func TestAppRequestResponse(t *testing.T) {
 	nodeID := ids.GenerateTestNodeID()
 	chainID := ids.GenerateTestID()
 
+	ctxKey := new(string)
+	ctxVal := new(string)
+	*ctxKey = "foo"
+	*ctxVal = "bar"
```

Comment:
why not
```
ctxKey := "foo"
ctxVal := "bar"
```
?

Reply:
This is because this fails linting, this was my original approach:

```
Error: network/p2p/router_test.go:108:37: SA1029: should not use built-in type string as key for value; define your own type to avoid collisions (staticcheck)
							ctx = context.WithValue(ctx, foo, bar)
```

We either have to make a struct or use a pointer. I used a pointer to avoid having to define my own type but that's an alternative we can do.